File: 프로그래머스/연습문제/Lv2/귤 고르기/귤 고르기.py
def solution(k, tangerine):
    tangerine = sorted(tangerine)

    cnts = []
    i = 0
    while i < len(tangerine):
        j = i+1
        while j<len(tangerine) and tangerine[i]==tangerine[j]:
            j += 1
        cnts.append(j-i)
        i = j

    cnts = sorted(cnts, reverse=True)

    cnt = 0
    for c in cnts:
        if k <= 0: break
        cnt += 1
        k -= c

    return cnt

File: 프로그래머스/연습문제/Lv2/귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(int k, vector<int> tangerine) {
    // 귤 크기대로 정렬
    sort(tangerine.begin(), tangerine.end());
    
    vector<int> cnts;
    for(int i=0; i<tangerine.size(); i++){
        int j;
        for(j=i+1; j<tangerine.size() && tangerine[i]==tangerine[j]; j++);
        cnts.push_back(j-i);
        i = j - 1;
    }
    
    // 개수 많은 순으로 정렬
    sort(cnts.begin(), cnts.end(), greater<int>());
    
    int cnt = 0;
    for(int c : cnts){
        if(k <= 0) break;
        cnt++;
        k -= c;
    }
    
    return cnt;
}
